Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm

```diff
@@ -1479,7 +1478,7 @@ - (void)URLSession:(NSURLSession *)session webSocketTask:(NSURLSessionWebSocketT
 
     configuration.get()._preventsSystemHTTPProxyAuthentication = parameters.preventsSystemHTTPProxyAuthentication;
     configuration.get()._requiresSecureHTTPSProxyConnection = parameters.requiresSecureHTTPSProxyConnection;
-    configuration.get().connectionProxyDictionary = (NSDictionary *)parameters.proxyConfiguration.get() ?: proxyDictionary(parameters.httpProxy, parameters.httpsProxy).get();
+    configuration.get().connectionProxyDictionary = RetainPtr<NSDictionary> { (NSDictionary*)parameters.proxyConfiguration.get() }.get() ?: proxyDictionary(parameters.httpProxy, parameters.httpsProxy).get();
```

Comment:
I would specially appreciate comments on this one, I am not sure this is the best approach or if it will suffice to address the warning.